### ISSUE TITLE
ByProgramModal displaying incorrect thresholdMOS

### ISSUE THREAD

**Issue by wlthomson** [openmsupply/mobile#1317]
## Describe the bug

When creating a new program requisition, the `orderType` in `modalProps` incorrectly (?) sets threshold text as `item.maxMOS`:

```
orderType: {
    ...
    renderRightText: item => {
      const { maxMOS, maxOPP, threshMOS } = getLocalizedStrings();
      const { maxOrdersPerPeriod: maxOrders, maxMOS: itemMOS } = item;
      const mosText = `${maxMOS}: ${item.maxMOS}`;
      const thresholdText = `${threshMOS}: ${itemMOS}`;
      const maxOrdersText = `${maxOPP}: ${maxOrders}`;
      return `${mosText} - ${thresholdText} - ${maxOrdersText}`;
    },
  },
```

### To reproduce

1. Create an order type with `Max MOS` !=  `Threshold MOS`
1. Go to `Supplier Requisitions`
2. Click on `New Requisition`.
3. Select a program and supplier.
4. Select an order type.
5. See error.

### Version and device info

- App version: v.2.3.6.
- Tablet model: Lenovo TB-X304L
- OS version: Android 8.1.0.

### Additional context

N/A.

**Comment by joshxg** [comment]
Closed by #1318

**Comment by joshxg** [comment]
# TESTING

- [x] Create a new program-based requisition for an order type with differing `Max MOS` and `Threshold MOS`. The `byProgramModal` correctly displays the threshold MOS.
